collectd: 5.7.2 -> 5.8.0

volth opened the pull request:
###### Motivation for this change

New version is aware about new versions of the softwares ```collectd``` can monitor (for example Varnish 5.2.x)

Fixes #32061

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

volth commented:
@grahamc What does "Failed to enumerate outputs of master" mean?

grahamc commented:
@GrahamcOfBorg eval

globin commented:
@GrahamcOfBorg build collectd

globin commented:
@GrahamcOfBorg eval

volth commented:
@grahamc something bad with CI Bot

grahamc commented:
I wrote a long comment and  github deleted it :(

@volth Thanks for the ping!

"Failed to enumerate outputs of master" means that some commit broke master, and we can't check this PR because master is broken. I'm wanting to improve that, but it is going to take a bit of time (https://github.com/grahamc/ofborg/issues/25) in the meantime I changed the error message to hopefully be a bit clearer: `Target branch master doesn't evaluate!` is that clearer? It will also provide a Details link with the evaluation error.

It looks like collectd might depend transitively on Java somehow? Right now we don't have a way to build java on aarch64 yet. I recently changed the builders to use --show-trace when building, so hopefully that'll provide more context to the error. I'd ignore the (unrelated) aarch64 error for now.

Does this make sense? PS: just in case, I'm going to try the build again.

grahamc commented:
@GrahamcOfBorg build collectd

grahamc commented:
^ LGTM

volth commented:
>  It looks like collectd might depend transitively on Java somehow? 

Yes, but it is optional.
Almost all of the ```collectd```'s dependencies are for its plugins and thus optional.

>  Right now we don't have a way to build java on aarch64 yet.

```oraclejdk8``` works on ```aarch64-linux```. 
It might have sense to set it as default ```jdk```/```jre``` for ```aarch64-linux``` while openjdk is broken.
Not only for ```collectd```, there are so many java apps (```vuze```, etc)

bjornfor commented:
Please squash commits.

volth commented:
@bjornfor Do not you have the "squash merge" button like ![](https://help.github.com/assets/images/help/pull_requests/select-squash-and-merge-from-drop-down-menu.png) ?

It is difficult to squash on my side, branches with names like ```patch-NN``` are created on Github website and I do not have them locally, while squash on the Github website is available only for repo owners.

bjornfor commented:
Yes, I have squash button. But I think maintaining a PR is the responsibility of the author. And I wouldn't like it if people squashed my PRs without asking.

What are those patch-NN branches? I've never seen them, and I submit PRs regularly. Amending or squashing a commit, at least in my experience, is just as easy as creating a new one. (Once you know how, of course.)

orivej commented:
`./configure` prints:
```
configure: WARNING: unrecognized options: --with-libsigrok, --with-python
```
This part needs updating.

(I don't mind reformatting the final commit or commits myself before merging, although it's always nice when this is unnecessary.)

volth commented:
I fixed some other deps (rieman, rdkafka, mongoc, ...), and cleaned ```all-packages.nix``` from the details of ```collectd<->libsigrok``` relationship.

orivej commented on pkgs/top-level/all-packages.nix:
```diff
@@ -7270,12 +7269,10 @@ with pkgs;
-    version = "0.3.0";
-    sha256 = "0l3h7zvn3w4c1b9dgvl3hirc4aj1csfkgbk87jkpl7bgl03nk4j3";
-  };
+  inherit (callPackage ../development/tools/libsigrok { })
```
Please use `callPackages`: b89b6b2a7b397a087149f01a9fd27876edfc6986

orivej commented on pkgs/top-level/all-packages.nix:
```diff
@@ -1504,7 +1504,6 @@ with pkgs;
 
   collectd = callPackage ../tools/system/collectd {
     libmysql = mysql.lib;
-    libsigrok = libsigrok-0-3-0; # not compatible with >= 0.4.0 yet
```
This is the preferred style: overriding versions in `all-packages` rather than specifying concrete versions in `default.nix`. Please keep it.

volth replied:
I do not think ```collectd``` is such a case when it should be preferred.

  * For: customization. unlike ```libmysql``` which might be replaced with ```mariadb.client``` ```libsigrok``` is not a param to be customized here, ```collectd``` requires an obsolete version and it is immanent to ```collectd```, not to the configuration.

  * Against: ```collectd``` is often customized by reducing list of params and thus reduce the closure size (which is about 1Gb with all plugin and its deps).

It is common to have in ```/etc/nixos/configuration.nix``` something like:
 ```nix
callPackage ../tools/system/collectd {
  jdk = null;
  /* .... dozen of other params but ```libsigrok``` is forgotten */
}
``` 
Forgotten ```libsigrok``` would result in picking up ```pkgs.libsigrok``` which is of unsupported version 0.5

if you insist on that style, then maybe the param ```libsigrok``` should be renamed to something unique in order to prevent its unexpected defaulting to ```pkgs.libsigrok```

orivej replied:
> `callPackage ../tools/system/collectd`

I have not seen this form of overriding before, I doubt that it is common, and it certainly should not be used. Overriding is documented in Nixpkgs manual: https://nixos.org/nixpkgs/manual/#sec-overrides . The correct form is `pkgs.collectd.override { jdk = null; }`.

Please, keep this line as is.

orivej commented:
Could you enable parallel building, and fix the build of `libcollectdclient`?
